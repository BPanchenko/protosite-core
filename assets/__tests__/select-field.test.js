import { Browser } from 'happy-dom'

const browser = new Browser()
const page = browser.newPage()

await page.goto('https://github.com/capricorn86')
