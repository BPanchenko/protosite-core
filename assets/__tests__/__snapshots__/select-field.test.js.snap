// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField Page Markup 1`] = `
"<!DOCTYPE html><html lang="ru"><head>
		<title>[TEST FORM OBJECT] Select Field Form ⁕ Protosite Core</title>
		<link href="http://assets.protosite.rocks/fonts/iconic/stylesheet.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/document.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/utilities.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/component/panel.css" rel="stylesheet" type="text/css">
		
		<script src="../select-field-component.mjs" type="module"></script>
		<style>
			.u-grid {
				display: grid;
				grid-template-columns: auto auto;
				gap: 8vmin;
				justify-content: space-around;
				align-items: center;
			}

			.c-panel:first-of-type .u-grid {
				block-size: 22dvh;
				margin-block-end: 8dvh;
			}
		</style>
	</head>
	<body class="t-document">
		<main class="u-container" data-testid="container">
			<section class="c-panel s-clean">
				<div class="u-grid">
					<c-select-field name="clr-copy-format" status-label="Копировать цвет" exportparts="button, listbox, value">
						<div role="option" data-value="hex">в формате HEX</div>
						<div role="option" data-value="rgb">в формате RGB</div>
						<div role="option" data-value="css-var">как переменную CSS</div>
						<div role="option" data-value="js-const">как константу JS</div>
					</c-select-field>
					<div>
						<label for="ice-cream-choice">Choose a flavor:</label>
						<input list="ice-cream-flavors" id="ice-cream-choice" name="ice-cream-choice">
						<datalist id="ice-cream-flavors">
							<option value="Chocolate"></option>
							<option value="Coconut"></option>
							<option value="Mint"></option>
							<option value="Strawberry"></option>
							<option value="Vanilla"></option>
						</datalist>
					</div>
				</div>
			</section>
			<section class="c-panel">
				<div class="u-grid">
					<c-select-field name="clr-copy-format" status-label="Копировать цвет" exportparts="button, listbox, value">
						<div role="option" data-value="hex">в формате HEX</div>
						<div role="option" data-value="rgb">в формате RGB</div>
						<div role="option" data-value="css-var">как переменную CSS</div>
						<div role="option" data-value="js-const">как константу JS</div>
					</c-select-field>
				</div>
			</section>
		</main>
		<template id="tpl-select-field">
			<input type="hidden" name="clr-copy-format">
			<div aria-activedescendant="" aria-controls="listbox" part="button" role="button">
				<div aria-label="" part="status" role="status"></div>
			</div>
			<slot name="listbox">
				<div aria-labelledby="??[role=button].id??" aria-multiselectable="off" role="listbox" part="listbox">
					<slot></slot>
				</div>
			</slot>
		</template>
	

</body></html>"
`;
